projects/transformers/trainer_mixins/multi_eval_sets.py

```diff
@@ -0,0 +1,85 @@
+# ----------------------------------------------------------------------
+# Numenta Platform for Intelligent Computing (NuPIC)
+# Copyright (C) 2020, Numenta, Inc.  Unless you have an agreement
+# with Numenta, Inc., for a separate license for this software code, the
+# following terms and conditions apply:
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+# See the GNU Affero Public License for more details.
+#
+# You should have received a copy of the GNU Affero Public License
+# along with this program.  If not, see http://www.gnu.org/licenses.
+#
+# http://numenta.org/licenses/
+# ----------------------------------------------------------------------
+
+
+class MultiEvalSetsTrainerMixin:
```

Comment:
This looks useful, to have it generalized. But introducing a class that handles the case of multiple eval sets begs the question of whether there is any use case except MLNI

Reply:
For now the only use case is MNLI. However, I think a mixin like this could be useful for other projects or different types of experiments (multi task learning, meta learning, continual learning). If you have a family of tasks, it could be informative to see how performance on each one changes over time, while training on a source dataset.